### PR TITLE
Streamline "addPublic*" parameters

### PR DESCRIPTION
**Pull request by kwin**
Extend Javadoc

### CODE REVIEW

**Review by mickaelistria — APPROVED** [review]

File: tycho-gpg-plugin/src/main/java/org/eclipse/tycho/gpg/SignRepositoryArtifactsMojo.java [hunk 1]
```diff
@@ -68,14 +72,22 @@ public class SignRepositoryArtifactsMojo extends AbstractGpgMojoExtension {
...
+     * If set to {@code true} adds the public key belonging to the signature to the repository
+     * metadata.
+     */
     @Parameter(defaultValue = "true")
     private boolean addPublicKeyToRepo;
 
-    @Parameter(defaultValue = "true")
-    private boolean addPublicKeysToArtifacts;
+    /**
+     * If set to {@code true} adds the public key belonging to the signature to each signed
```
Comment (lppedd) [inline tycho-gpg-plugin/src/main/java/org/eclipse/tycho/gpg/SignRepositoryArtifactsMojo.java:83]: Is `{@code ...}` handled correctly in the generated docs? @kwin

**Comment by laeubi** [comment]
I don't think adding a link to the eclipse-platform(!) documentation is the right choice here. This describes a very specific usecase that is most probably not applicable to most projects outside the eclipse world.

**Comment by mickaelistria** [comment]
> This describes a very specific usecase that is most probably not applicable to most projects outside the eclipse world

It's p2 documentation, and PGP signatures are part of p2 metadata. Tycho has no use case with PGP signature in p2 metadata that makes sense outside of p2 world, so it seems very fine to reference the upstream/target documentation.

**Comment by laeubi** [comment]
> It's p2 documentation, and PGP signatures are part of p2 metadata.

The page says

> **Eclipse** documentation - Current Release **Eclipse IDE 2022-09**

Further on

> **In Eclipse applications**, any plug-in can define some PGP public keys as trusted using the [org.eclipse.equinox.p2.engine.pgp](https://help.eclipse.org/latest/topic/org.eclipse.platform.doc.isv/reference/extension-points/org_eclipse_equinox_p2_engine_pgp.html) extension-point. Users can review and manage trusted keys in the **Opens the Trust preference pageInstall/Update > Trust preference page**.

So this all indicates (to me) that this is very specific on how Eclipse works and other application might behave differently, especially in terms of UI, key management and alike.

**Comment by mickaelistria** [comment]
I'm inclined to merge it as it and replace the link to another one when a better one is available.

**Comment by laeubi** [comment]
Just wanted to mention that the link is more how Eclipse (IDE) uses this currently than a generic one, but if you think its fine I won't veto it.

**Pull request merged**